### PR TITLE
Fixes #36723 - Remove Fedora/Red Hat Atomic related code

### PR DESCRIPTION
**Pull request by ekohl**
As far as I can see this has been EOL for a few years now. Red Hat Atomic went EOL in 2021.

### CODE REVIEW

**Comment by stejskalleos** [comment]
@nofaralfasi

File: app/services/katello/rhsm_fact_parser.rb [hunk 1]
```diff
@@ -146,8 +143,6 @@ def get_rhsm_mac(interface)
     end
 
     def distribution_to_puppet_os(name)
-      return REDHAT_ATOMIC_HOST_OS if name == REDHAT_ATOMIC_HOST_DISTRO_NAME
-
       name = name.downcase
```
Comment (ekohl) [inline app/services/katello/rhsm_fact_parser.rb]: I noticed this block could be cleaner using a case statement: https://github.com/theforeman/foreman/pull/9838

**Comment by nofaralfasi** [comment]
Hi @ekohl, can you please resolve the merge conflicts to I can rebase [this PR](https://github.com/theforeman/foreman/pull/9837) onto it?

**Comment by stejskalleos** [comment]
@ekohl any updates here? it's blocking the https://github.com/theforeman/foreman/pull/9837 which is for upcoming release and would be nice to have it ready for testing ASAP

**Review by stejskalleos — APPROVED** [review]
:green_apple: **LGTM**

**Comment by stejskalleos** [comment]
Thanks @ekohl

**Comment by ekohl** [comment]
Release note: https://github.com/theforeman/foreman-documentation/pull/2486

**Pull request merged**